akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/Effect.scala

```diff
@@ -58,17 +54,62 @@ object Effect {
  * Not for user extension.
  */
 @DoNotInherit
-trait Effect[+Event, State] { self: EffectImpl[Event, State] ⇒
+trait Effect[+Event, State] {
   /* All events that will be persisted in this effect */
   def events: im.Seq[Event]
 
-  def sideEffects[E >: Event]: im.Seq[ChainableEffect[E, State]]
-
-  /** Convenience method to register a side effect with just a callback function */
+  /**
+   * Run the given callback. Callbacks are run sequentially.
```

Comment:
I think this could be more clear. "Run the given callback once this effect has completed successfully."?